topdown/topdown_partial_test.go

```diff
@@ -3288,6 +3288,28 @@ func TestTopDownPartialEval(t *testing.T) {
 					a2 = input; __local4__2 > __local2__2 }
 				}`},
 		},
+		{ // https://github.com/open-policy-agent/opa/issues/5367
+			note:  "copypropagation: keep equasion that are only found in comprehensions, inlined function call",
+			query: "data.test.p",
+			modules: []string{`package test
+			key_exists(obj, k) { x = obj[k] }
```

Comment:
NB: I've called it `x` instead of `_` because otherwise I wouldn.t have been able to write out the query below -- no two `_` are alike 😅 -- but for topdown, it doesn't matter much.